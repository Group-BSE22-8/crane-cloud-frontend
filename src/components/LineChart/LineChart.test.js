import React from "react";
import { shallow } from "enzyme";
import LineChart from ".";

describe("Test the line chart component", () => {
  const linechartData = [
    { name: "A", min: 400, max: 2400, sum: 2400 },
    { name: "B", min: 40, max: 240, sum: 240 },
  ];
  const linechartPropsNoPreview = {
    data: linechartData,
    preview: false,
    xLabel: "",
    yLabel: "",
    xDataKey: "",
    yDataKey: "",
    lineDataKey: "",
  };
  const linechartPropsPreivew = {...linechartPropsNoPreview, preview:true};
  const linechartNoData = {...linechartPropsNoPreview, data:[]};

  const LineChartNoPreview = shallow(<LineChart {...linechartPropsNoPreview} />);
  const LineChartPreview = shallow(<LineChart {...linechartPropsPreivew} />);
  const LineChartNoData = shallow(<LineChart {...linechartNoData} />);
  it("checks if the information bar component matches the snapshot", () => {
    expect(LineChartNoPreview).toMatchSnapshot();
  });
});
