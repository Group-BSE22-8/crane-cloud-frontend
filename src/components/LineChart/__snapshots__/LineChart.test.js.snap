// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the line chart component checks if the information bar component matches the snapshot 1`] = `ShallowWrapper {}`;
